### ISSUE TITLE
cross-encoder not improving accuracy in semantic search

### ISSUE THREAD

**Issue by gypsydave5** [sinanuozdemir/quick-start-guide-to-llms#26]
Hey! I'm coding along with the course and I thought I might have made some mistake - the semantic search results re-ranked with the cross encoder have the same accuracy as the results before re-ranking.

So I pulled the notebook from this repo to run it... and it's also the same.

Has one of the models changed under the hood in the last six months? Is there anything I could try out to see a cross-encoder at work?

(enjoying the book + course otherwise - ty!)

**Comment by sinanuozdemir** [comment]
Hey David, thanks so much for letting me know! Just to be certain, you're talking about numbers from [02_semantic_search.ipynb](https://github.com/sinanuozdemir/quick-start-guide-to-llms/blob/main/notebooks/02_semantic_search.ipynb), right? It's showing that accuracy went up after re-ranking but I'll have to re-run the code myself when I get the chance to see what may have gone wrong here!


In the meantime, here's another example of re-ranking from a RAG course I give [here](https://github.com/sinanuozdemir/oreilly-retrieval-augmented-gen-ai/blob/main/notebooks/RAG_Retrieval.ipynb). I hope this helps! It's using a closed-source re-ranker though just a heads up, not open-source.

I really do appreciate you bringing this to my attention!

**Comment by sinanuozdemir** [comment]
I fixed the issue and the code should be running correctly now! I even took the opportunity to improve how I created training data for the fine-tuning section and got even better results from the fine-tuned cross encoder. I hope this resolves the issue, David! Thanks again for making the issue.

**Comment by gypsydave5** [comment]
Ahhhh - thank you so much @sinanuozdemir - I thought I was going mad. Really appreciate the responsiveness. All the best!
